instrumentation/http-tests/src/main/java/brave/test/http/ITServlet3Container.java

```diff
@@ -30,59 +34,88 @@
 public abstract class ITServlet3Container extends ITServlet25Container {
   static ExecutorService executor = Executors.newCachedThreadPool();
 
-  @AfterClass
-  public static void shutdownExecutor() {
-    executor.shutdownNow();
+  public ITServlet3Container() {
+    super(new Jetty9ServerController());
   }
 
-  static class AsyncServlet extends HttpServlet {
-    @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
-      if (Tracing.currentTracer().currentSpan() == null) {
-        throw new IllegalStateException("couldn't read current span!");
-      }
-      AsyncContext ctx = req.startAsync();
-      ctx.start(ctx::complete);
-    }
+  @AfterClass public static void shutdownExecutor() {
+    executor.shutdownNow();
   }
 
-  @Test
-  public void forward() throws Exception {
+  @Test public void forward() throws Exception {
     get("/forward");
 
     takeSpan();
   }
 
-  @Test
-  public void forwardAsync() throws Exception {
+  @Test public void forwardAsync() throws Exception {
     get("/forwardAsync");
 
     takeSpan();
   }
 
   static class ForwardServlet extends HttpServlet {
-    @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+    @Override protected void doGet(HttpServletRequest req, HttpServletResponse resp)
       throws ServletException, IOException {
       req.getServletContext().getRequestDispatcher("/foo").forward(req, resp);
     }
   }
 
   static class AsyncForwardServlet extends HttpServlet {
-    @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+    @Override protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
       AsyncContext asyncContext = req.startAsync(req, resp);
       executor.execute(() -> asyncContext.dispatch("/async"));
     }
   }
 
+  static class AsyncServlet extends HttpServlet {
+    @Override protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+      if (Tracing.currentTracer().currentSpan() == null) {
+        throw new IllegalStateException("couldn't read current span!");
+      }
+      AsyncContext ctx = req.startAsync();
+      ctx.start(ctx::complete);
+    }
+  }
+
   static class ExceptionAsyncServlet extends HttpServlet {
-    @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+    @Override protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+      if (DispatcherType.ERROR.equals(req.getDispatcherType())) return; // don't loop
+
+      AsyncContext async = req.startAsync();
+      // unless we add a listener, the onError hook will never occur
```

Comment:
I have seen some tests out there that say they are testing onError when they are really testing timeout. beware!

below, I prove onError works by switching the http status